### ISSUE TITLE
Setting `typeName: null` from appsettings.json

### ISSUE THREAD

**Issue by snebjorn** [serilog-contrib/serilog-sinks-elasticsearch#475]
**Does this issue relate to a new *feature* or an existing *bug*?**
- [x] Bug
- [ ] New Feature

**What version of Serilog.Sinks.Elasticsearch is affected?  Please list the related NuGet package.**
serilog.sinks.elasticsearch v8.4.1

**What is the target framework and operating system? See [target frameworks](https://docs.microsoft.com/en-us/nuget/schema/target-frameworks) & [net standard matrix](https://docs.microsoft.com/en-us/dotnet/standard/net-standard).**

- [ ] netCore 2.0
- [ ] netCore 1.0
- [ ] 4.7
- [ ] 4.6.x
- [ ] 4.5.x

**Please describe the current behavior?**
Given this appsettings.json 
```jsonc
{
  "Serilog": {
    "Using": ["Serilog.Sinks.Elasticsearch"],
    "WriteTo": {
      "ElasticsearchSink": {
        "Name": "Elasticsearch",
        "Args": {
          "nodeUris": "...",
          "typeName": null
        }
      }
    }
  }
}
```
then the `typeName` isn't set to `null` in the config.

**Please describe the expected behavior?**

The expected behavior is that when `"typeName": null` is set on the ElasticsearchSink appsettings.json config it should configure the `TypeName` to be `null`.

**If the current behavior is a bug, please provide the steps to reproduce the issue and if possible a minimal demo of the problem**

Given the above config, load it using the **Serilog.Settings.Configuration** extension

```cs
.UseSerilog(
    (context, services, configuration) =>
    {
        configuration.ReadFrom.Configuration(context.Configuration);
    }
)
```

**Comment by Mfarsi138** [comment]
i have same problem

**Comment by kamilzima92** [comment]
Try this (setup Typename:null and MiniumLogEvent: (i tried Verbose):
`
"WriteTo": [
      {
        "Name": "Elasticsearch",
        "Args": {
          "nodeUris": "https://elastic:password@127.0.0.1:9200",
          "indexFormat": "JSON-index-{0:yyyy.MM}",
          "typeName": null,
          "MinimumLogEventLevel": "Verbose"
        }
      }
    ]`

Should Work

**Comment by zachawilson** [comment]
It looks like this is included in the beta release `v9.0.0-beta7`

- https://github.com/serilog-contrib/serilog-sinks-elasticsearch/discussions/443
- https://github.com/serilog-contrib/serilog-sinks-elasticsearch/pull/420

**Comment by nenadvicentic** [comment]
This has been fixed in `Serilog.Sinks.Elasticsearch` `v9.0.0`.

Also, note that, when `ElasticsearchSinkOptions.DetectElasticsearchVersion` is set to `true` (which is now default setting), `options.TypeName` will be ignored when Elasticsearch 7 or higher is detected).

For Elasticsearch server `v7`, the sink will use mandatory type name "_doc", while for `v8+` type name is going to be omitted.

The issue can be closed.
